### DISCUSSION TITLE
package.json → package.jsonc

### DISCUSSION THREAD

**Discussion by charnould** [oven-sh/bun#9036] [Ideas]
A huge pain in JS world is you cannot add comments into `package.json`.
**What if `Bun` allows it?** It would be fantastic (to me) and a HUGE advantage over `Node` :)

I've tried to change `package.json` to `package.jsonc` and `Bun` returns an error :-(

```
error: MissingPackageJSON
Bun could not find a package.json file.
```

PS. Thanks for making `Bun`

**Comment by truongan07** [comment]
Yes, i think this is huge advantage to use comments in config file. We can describe keys

**Comment by KilianB** [comment]
A very common use case for me is the need to comment why dependencies are pinned to a specific version e.g. due to a bug or dependency of some other package. For this reason alone I would  really enjoy this feature. 

I would assume that implementing this feature is going to be rather tricky because so much tooling is using the package.json convention. Maybe we can have an additional step which just generates a pckage.json on the fly?, is the impact of the startup time to severe?

**Reply by truongan07** [reply]
We only need comments in jsonc file so importing jsonc will remove all comments to return a json string and parse it to get object

**Reply by charnould** [reply]
In my mind it's as "simple" as @truongan07 says it (:-p). 
Maybe not (I cannot write a single line of a runtime), but it would be awesome to have just a package.jsonc!

**Reply by KilianB** [reply]
This aged well, see the last comments about vitest not running. I tried to publish a package to npm and esbuild? was bugging out. 

I really lobby for comments inside `package.json` and would use it every single time,  the issue is just that as soon as the project gets a tad bit more complicated with different tools it will break for sure. Maybe at some point bun will be able to provide the package.json at runtime and supports having a second package.jsonc which gets synched after a change?

**Comment by wendevlin** [comment]
Hi,
I wanted something similar with yaml and created an issue: https://github.com/oven-sh/bun/issues/7468

But it was closed and I get following anwser from Jarred: 

> I definitely agree that json isn't perfect. It's missing features like comments and trailing commas. But syntax doesn't add enough > value to justify the ecosystem cost of a decision like this. You could use a build tool that converts your package.toml or package.yaml into another format ahead of time into a package.json though

**Comment by hatton** [comment]
This would be _the thing_ that would switch my team from nodejs to bun.

**Comment by truongan07** [comment]
resolved in v.1.1.5

```
import jsonc from './config/services.config.jsonc' with { type: 'text' };
```

**Comment by kristian-nst** [comment]
Agree! Rly want comments in the package.json
All these discussions deny supporting it for "not worth it for the ecosystem changing".

But devs have been asking about it for probably a decade already.
Why can't "we" (bun) be the first to do so? Let's move forward **now** so the developers for the **next decade** can have something nice :) 🚀 

The future devs should laugh at us not having comment support :P Isn't this how software evolves?
JSONC isn't even something completely new, it already exists and is a "solved" thing.

**Comment by Jarred-Sumner** [comment]
You can add comments and trailing commas to package.json in Bun. We have test coverage for this. The filename continues to be package.json.

**Reply by kristian-nst** [reply]
Maybe it "works", but almost every editor will show you errors/warnings about it.

![image](https://github.com/user-attachments/assets/f6f99103-8b53-4bc2-a15d-1613689cef6d)

the file extension is important..

**Reply by vicb** [reply]
VSCode? you can associate package.json with the jsonc content type.

**Reply by kristian-nst** [reply]
that only works per machine, not rly convenient across a big team

**Reply by aster-void** [reply]
https://code.visualstudio.com/docs/getstarted/settings#_workspace-settingsjson-location

a package manager isn't responsible for the file looking nice in your IDE, and I think supporting both extension introduces too much complexity

**Reply by charnould** [reply]
As @aster-void said, using vscode `settings.json` solved warning in IDE. However, in my cas, `prettier` throws. [It seems to be a `prettier` limit](https://github.com/prettier/prettier/issues/17039), but have any of you already found a workaround?

**Reply by zemjr** [reply]
We can't use vitest as it does not support comments in the package.json too

**Comment by MeshyIce** [comment]
Hey, for anyone excited about using comments, please note that comments in your `package.json` file will be removed after external modifications from bun (such as after `bun install`ing a new package)

See https://github.com/oven-sh/bun/issues/11536

**Reply by agrublev** [reply]
This is literally why i built https://fscr.one :) so you can have beautiful **Markdown** documents which are actually executable 💚💚💚
